hardwareModules/RoFICoM/software/i2c_test/src/main.cpp

```diff
@@ -0,0 +1,193 @@
+
+    /* Set APB1 prescaler*/
+    LL_RCC_SetAPB1Prescaler( LL_RCC_APB1_DIV_1 );
+
+    LL_Init1msTick(64000000);
+
+    LL_SYSTICK_SetClkSource( LL_SYSTICK_CLKSOURCE_HCLK );
+    /* Update CMSIS variable (which can be updated also through SystemCoreClockUpdate function) */
+    LL_SetSystemCoreClock( 64000000 );
+    LL_RCC_SetUSARTClockSource( LL_RCC_USART1_CLKSOURCE_PCLK1 );
+    LL_RCC_SetUSARTClockSource( LL_RCC_USART2_CLKSOURCE_PCLK1 );
+}
+
+void setupI2C() {
+    LL_GPIO_InitTypeDef GPIO_InitStruct = { };
+
+    LL_IOP_GRP1_EnableClock(LL_IOP_GRP1_PERIPH_GPIOA);
+    /**I2C2 GPIO Configuration
+    PA11 [PA9]   ------> I2C2_SCL
+    PA12 [PA10]   ------> I2C2_SDA
+    */
+    GPIO_InitStruct.Pin = LL_GPIO_PIN_11;
+    GPIO_InitStruct.Mode = LL_GPIO_MODE_ALTERNATE;
+    GPIO_InitStruct.Speed = LL_GPIO_SPEED_FREQ_LOW;
+    GPIO_InitStruct.OutputType = LL_GPIO_OUTPUT_OPENDRAIN;
+    GPIO_InitStruct.Pull = LL_GPIO_PULL_UP;
+    GPIO_InitStruct.Alternate = LL_GPIO_AF_6;
+    LL_GPIO_Init(GPIOA, &GPIO_InitStruct);
+
+    GPIO_InitStruct.Pin = LL_GPIO_PIN_12;
+    GPIO_InitStruct.Mode = LL_GPIO_MODE_ALTERNATE;
+    GPIO_InitStruct.Speed = LL_GPIO_SPEED_FREQ_LOW;
+    GPIO_InitStruct.OutputType = LL_GPIO_OUTPUT_OPENDRAIN;
+    GPIO_InitStruct.Pull = LL_GPIO_PULL_UP;
+    GPIO_InitStruct.Alternate = LL_GPIO_AF_6;
+    LL_GPIO_Init(GPIOA, &GPIO_InitStruct);
+
+    /* Peripheral clock enable */
+    LL_APB1_GRP1_EnableClock(LL_APB1_GRP1_PERIPH_I2C2);
+
+    LL_I2C_InitTypeDef I2C_InitStruct = { };
+
+    I2C_InitStruct.PeripheralMode = LL_I2C_MODE_I2C;
+    // This parameter is computed by STM32cubemx tool
+    I2C_InitStruct.Timing = 0x00303D5B;
+    I2C_InitStruct.AnalogFilter = LL_I2C_ANALOGFILTER_ENABLE;
+    I2C_InitStruct.DigitalFilter = 0;
+    I2C_InitStruct.OwnAddress1 = 0;
+    I2C_InitStruct.TypeAcknowledge = LL_I2C_ACK;
+    I2C_InitStruct.OwnAddrSize = LL_I2C_OWNADDRESS1_7BIT;
+
+    LL_I2C_Init(I2C2, &I2C_InitStruct);
+    LL_I2C_EnableAutoEndMode(I2C2);
+    LL_I2C_SetOwnAddress2(I2C2, 0, LL_I2C_OWNADDRESS2_NOMASK);
+    LL_I2C_DisableOwnAddress2(I2C2);
+    LL_I2C_DisableGeneralCall(I2C2);
+    LL_I2C_EnableClockStretching(I2C2);
+}
+
+
+using Block = memory::Pool::Block;
+
+Dbg& dbgInstance() {
+    static Dbg inst(
+        USART1, Dma::allocate( DMA1, 1 ), Dma::allocate( DMA1, 2 ),
+        TxOn( GpioB[ 6 ] ),
+        RxOn( GpioB[ 7 ] ),
+        Baudrate( 115200 ) );
+    return inst;
+}
+
+// Helper class for printing error on `Result<Void, std::string>`
+class PrintError {
+    public:
+        void operator()( std::true_type, auto ) { }
+        void operator()( std::false_type, const std::string& error ) { Dbg::error( error.c_str() ); }
+};
+
+void LidarMeasure() {
+    Lidar lidar( I2C( I2C2, SdaPin( GpioA[ 12 ] ), SclPin( GpioA[ 11 ] ) ) );
+
+    Dbg::info("VL53L1 init start\n");
+    PrintError printError;
+    lidar.initialize().match( printError );
```

Comment:
Chápu, že se jedná o testovací proof-of-concept firmware, takže to nevadí, ale chci abychom na to nezapomněli - je potřeba umět na chybu reagovat a třeba zabránit dalším měřením).